### DISCUSSION TITLE
How to create a DAG workflow that uses a single container (Without bootstrapping per-task containers)

### DISCUSSION THREAD

**Discussion by xahon** [argoproj/argo-workflows#8705] [Q&A] [answered]
I want to reduce the performance overhead of creating multiple containers for each task in a DAG and run all tasks in a single container. Is it possible?

**Comment by terrytangyuan** [comment, marked as answer]
Do you mean multiple tasks in a single pod? If so, then you can use container set feature: https://github.com/argoproj/argo-workflows/blob/master/docs/container-set-template.md

If you want to run DAG in a single container, then Argo Workflows might not be a good fit for you.
